packages/core/src/actors/callback.ts

```diff
@@ -134,36 +134,25 @@ export function fromCallback<TEvent extends EventObject, TInput = unknown>(
 ): CallbackActorLogic<TEvent, TInput> {
   const logic: CallbackActorLogic<TEvent, TInput> = {
     config: invokeCallback,
-    start: (_state, { self, system }) => {
-      system._relay(self, self, { type: 'xstate.create' });
-    },
-    transition: (state, event, { self, system }) => {
-      if (event.type === 'xstate.create') {
-        const sendBack = (eventForParent: AnyEventObject) => {
-          if (state.status === 'stopped') {
+    start: (state, { self, system }) => {
+      state._dispose = invokeCallback({
+        input: state.input,
+        system,
+        self,
+        sendBack: (event) => {
+          if (self.getSnapshot().status === 'stopped') {
```

Comment:
This line looks like an actual bug fix - it looks like we were using stale closure data here before. A test case for this should be added